providers/base/bin/reboot_check_test.py

```diff
@@ -275,42 +274,133 @@ def has_display_connection(self) -> bool:
 
     def is_hardware_renderer_available(self) -> bool:
         """
-        Checks if hardware rendering is being used.
-        THIS ASSUMES A DRM CONNECTION EXISTS
+        Checks if hardware rendering is being used by calling glmark2
+        - THIS ASSUMES A DRM CONNECTION EXISTS AND ALL ENVS HAVE BEEN SET
         - self.has_display_connection() should be called first if unsure
 
         :return: True if a hardware renderer is active, otherwise return False
         :rtype: bool
         """
 
-        DISPLAY = os.getenv("DISPLAY", "")
+        DISPLAY = os.getenv("DISPLAY")
+        XDG_SESSION_TYPE = os.getenv("XDG_SESSION_TYPE")
 
-        if DISPLAY == "":
-            print("$DISPLAY is not set, we will let unity_support infer this")
-        else:
-            print("Checking $DISPLAY={}".format(DISPLAY))
+        if not DISPLAY:
+            print("$DISPLAY is not set, marking the test as failed")
+            return False
 
-        unity_support_output = sp.run(
-            ["{}/usr/lib/nux/unity_support_test".format(RUNTIME_ROOT), "-p"],
-            stdout=sp.PIPE,
-            universal_newlines=True,
+        if not XDG_SESSION_TYPE:
+            print("$XDG_SESSION_TYPE is not set, marking the test as failed")
+            return False
+
+        if XDG_SESSION_TYPE not in ("x11", "wayland"):
+            # usually it's tty if we get here,
+            # happens when gnome failed to start or not using graphical session
+            print(
+                "Unsupported session type: {}".format(XDG_SESSION_TYPE),
+                file=sys.stderr,
+            )
+            return False
+
+        print(
+            "Checking hardware renderer with these env variables:",
+            "DISPLAY={}".format(DISPLAY),
+            "XDG_SESSION_TYPE={}".format(XDG_SESSION_TYPE),
         )
-        if unity_support_output.returncode != 0:
+
+        cpu_arch = sp.check_output(
+            ["uname", "-m"], universal_newlines=True
+        ).strip()
+
+        if cpu_arch in ("x86_64", "amd64"):
+            # x86 DUTs should run the version that uses the full opengl api
+            glmark2_executable = "glmark2"
+        else:
+            # TODO: explicitly check for aarch64?
+            glmark2_executable = "glmark2-es2"
+
+        if XDG_SESSION_TYPE == "wayland":
+            glmark2_executable += "-wayland"
+        # if x11, don't add anything
+
+        try:
+            if RUNTIME_ROOT:
+                # the official way to specify the location of the data files
+                # is "--data-path path/to/data/files"
+                # but 16, 18, 20 doesn't have this option
+                # and the /usr/share/glmark2 is hard-coded inside glmark2
+                os.symlink(
+                    "{}/usr/share/glmark2".format(RUNTIME_ROOT),
+                    "/usr/share/glmark2",
+                    target_is_directory=True,
+                )
```

Comment:
This is a bit of a hack that I'm a bit worried about since it symlinks a file inside the snap. Is this considered bad practice inside checkbox?

Reply:
and the /usr/share/glmark2 path is a build time environment variable so we also can't change this by just specifying a different env